### PR TITLE
Add ADE20K and COCO v2 dataset behind a version flag

### PR DESCRIPTION
**Pull request by karan6181**
## Description 
- Add a ADE20K and COCO v2 dataset behind a version flag
- Old PR with a similar change for reference https://github.com/mosaicml/composer/pull/1507

### CODE REVIEW

File: composer/datasets/ade20k_hparams.py [hunk 1]
```diff
@@ -176,15 +179,61 @@ class StreamingADE20kHparams(DatasetHparams):
...
+                                      batch_size=batch_size)
+        elif self.version == 2:
+            try:
+                from streaming.vision import ADE20K
+            except ImportError as e:
+                raise MissingConditionalImportError(extra_deps_group='streaming',
+                                                    conda_package='mosaicml-streaming') from e
+            # Define data transformations based on data split
+            if self.split == 'train':
+                both_transforms = torch.nn.Sequential(
```
Comment (knighton) [inline composer/datasets/ade20k_hparams.py:200]: should this be `torchvision.transforms.Compose`

File: composer/datasets/ade20k_hparams.py [hunk 2]
```diff
@@ -176,15 +179,61 @@ class StreamingADE20kHparams(DatasetHparams):
...
+                        class_max_percent=0.75,
+                        num_retry=10,
+                    ),
+                    RandomHFlipPair(),
+                )
+
+                # Photometric distoration values come from mmsegmentation:
+                # https://github.com/open-mmlab/mmsegmentation/blob/aa50358c71fe9c4cccdd2abe42433bdf702e757b/mmseg/datasets/pipelines/transforms.py#L861
+                r_mean, g_mean, b_mean = IMAGENET_CHANNEL_MEAN
+                image_transforms = torch.nn.Sequential(
```
Comment (knighton) [inline composer/datasets/ade20k_hparams.py:215]: should this be `torchvision.transforms.Compose`

**Review by knighton — APPROVED** [review]

File: composer/datasets/ade20k_hparams.py (hunk 1 above)
Comment (karan6181) [reply composer/datasets/ade20k_hparams.py:200]: Generally we don’t use `torch.nn.Sequential()` for non-scriptable transformation and `transforms.ToTensor()` is not a scriptable transformation and we don’t have that in ADE20K. So, it's fine to use  `torch.nn.Sequential()` here.  Also, the code structure is similar to existing [ade20k.py](https://github.com/mosaicml/composer/blob/dev/composer/datasets/ade20k.py#L339).

File: composer/datasets/ade20k_hparams.py (hunk 2 above)
Comment (karan6181) [reply composer/datasets/ade20k_hparams.py:215]: Generally we don’t use `torch.nn.Sequential()` for non-scriptable transformation and `transforms.ToTensor()` is not a scriptable transformation and we don’t have that in ADE20K. So, it's fine to use  `torch.nn.Sequential()` here.  Also, the code structure is similar to existing [ade20k.py](https://github.com/mosaicml/composer/blob/dev/composer/datasets/ade20k.py#L339).

**Pull request merged**